frontend/src/__tests__/cypress/cypress/e2e/customServingRuntimes/CustomServingRuntimes.cy.ts

```diff
@@ -73,38 +92,239 @@ describe('Custom serving runtimes', () => {
+    ]);
+    servingRuntimes.findSelectServingPlatformButton().click();
+
+    servingRuntimes.findSubmitButton().should('be.disabled');
     servingRuntimes.shouldSelectPlatform('Multi-model serving platform');
     servingRuntimes.findSelectAPIProtocolButton().should('not.be.enabled');
     servingRuntimes.findSelectAPIProtocolButton().should('include.text', 'REST');
     servingRuntimes.findStartFromScratchButton().click();
-    servingRuntimes.getDashboardCodeEditor().findInput().type('test');
-    servingRuntimes.findCreateButton().should('be.enabled');
+    servingRuntimes.uploadYaml(addfilePath);
+    servingRuntimes.findSubmitButton().should('be.enabled');
+    servingRuntimes.findSubmitButton().click();
+
+    cy.wait('@createMultiModelServingRuntime').then((interception) => {
+      expect(interception.request.url).to.include('?dryRun=All');
+      expect(interception.request.body.metadata).to.eql({
+        name: 'template-new',
+        annotations: { 'openshift.io/display-name': 'New OVMS Server' },
+        labels: { 'opendatahub.io/dashboard': 'true' },
+        namespace: 'opendatahub',
+      });
+    });
+
+    cy.wait('@createTemplate').then((interception) => {
+      expect(interception.request.body.metadata.annotations).to.eql({
+        'opendatahub.io/modelServingSupport': '["multi"]',
+        'opendatahub.io/apiProtocol': 'REST',
+      });
+      expect(interception.request.body.objects[0].metadata).to.eql({
+        name: 'template-new',
+        annotations: { 'openshift.io/display-name': 'New OVMS Server' },
+        labels: { 'opendatahub.io/dashboard': 'true' },
+      });
+    });
   });
 
   it('should duplicate a serving runtime', () => {
+    cy.intercept(
+      {
+        method: 'POST',
+        pathname: '/api/servingRuntimes',
+      },
+      mockServingRuntimeTemplateK8sResource({}).objects,
+    ).as('duplicateServingRuntime');
+    cy.intercept(
+      {
+        method: 'POST',
+        pathname: '/api/templates/',
+      },
+      mockServingRuntimeTemplateK8sResource({}),
+    ).as('duplicateTemplate');
+
+    const ServingRuntimeTemplateMock = mockServingRuntimeTemplateK8sResource({
+      name: 'serving-runtime-template-1',
+      displayName: 'Multi platform',
+      platforms: [ServingRuntimePlatform.SINGLE],
+      apiProtocol: ServingRuntimeAPIProtocol.GRPC,
+    });
+
+    cy.intercept(
+      { pathname: '/api/templates/opendatahub' },
+      mockK8sResourceList([...initialMock, ServingRuntimeTemplateMock]),
+    ).as('refreshServingRuntime');
+
     servingRuntimes.getRowById('template-1').find().findKebabAction('Duplicate').click();
     servingRuntimes.findAppTitle().should('have.text', 'Duplicate serving runtime');
+    cy.url().should('include', '/addServingRuntime');
+
+    servingRuntimes.shouldDisplayAPIProtocolValues([
+      ServingRuntimeAPIProtocol.REST,
+      ServingRuntimeAPIProtocol.GRPC,
+    ]);
+    servingRuntimes.shouldSelectAPIProtocol(ServingRuntimeAPIProtocol.GRPC);
```

Comment:
Please rename `shouldSelectAPIProtocol` to `selectAPIProtocol` as the `should` prefix is used only for assertion functions.